accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,361 @@
+  (could also be "meta-templated" by SUSE Manager's "activation keys"
+  that define, channels, admin users, groups)
+
+- Boot if necessary
+
+- Apply highstate, including yomi state
+
+2) Pre-populate the DB:
+
+- Create a new "empty" node in the DB and provide the data needed to
+  match it with a physical system (e.g. MAC address or serial number)
+
+- Apply custom configuration by adding formulas and filling in forms
+  (could also be "meta-templated" by SUSE Manager's "activation keys"
+  that define, channels, admin users, groups)
+
+- Boot node (via PXE, UEFI network boot, ISO)
+
+- Register it in SUSE Manager and populate the hardware data
+
+- Apply highstate, including yomi state
+
+(1) exists as a feature, but currently the "discovery image" isn't
+using Salt yet. We could re-use the exact same image that you're using
+for installation, just stop before applying the yomi state.
+
+(2) also exists for saltboot. But as of today it's just an API call,
+no UI version yet. Trivial to add. What's missing is applying an
+activation key to such a system as a template. As of today you'd have
+to inject the activation key grain on the minion side before
+registering the system to Manager.
+
+--<--<--
+
+## Disabling automatic highstate
+
+If I understand the issue correctly, the fact that the Yomi pillar is
+split into multiple sub-forms makes the application of partially
+defined pillar data more probable that if it is defined in a single,
+big form, without defaults.
+
+To potentially fix this issue, I propose in this RFC to have a grain
+emitted by the just-discovered system that will disable any automatic
+highstate application, if present. Only the highstate for this system
+can be manually applied:
+
+```yaml
+  automatic-highstate: no
+```
+
+## High-level forms?
+
+Another initial problem is that the pillar data is maybe too low-level
+for a typical use case. The pillar requires the list of packages that
+will be installed, repositories that will be registered, list of
+subvolumes for a BtrFS file system, use or not of snapper, partition
+layout, etc. You can express MicroOS or classical installations with
+those parameters, or create new mixtures of them tailored for specific
+use cases that are not directly supported by SUSE.
+
+A, maybe, better approach can be to provide some UI where the user can
+select the typical installation profile, similar to the skelcd-control
+from YaST. Based on that and the hardware information, a proposal of a
+pillar can be generated for this system.
+
+Is not clear to me how we can integrate high-level and low-level forms
+inside Uyuni, without making very confusing the usage of the
+interface. For example, if we decide to register the repositories from
+SLE via a high-level form, maybe is a good idea to provide a mechanism
+that will disable the low-level version of the form where the
+repositories and packages can be registered, to avoid possible
+conflicts.
+
```

Comment:
Formulas assigned to an individual system can already override the defaults from the group. But we know that finding out where things are coming from is a bit messy and needs some re-work. Not just for this use case.

Reply:
I added a note about saltclass. I think that can cover this case too.